tests/components/openhome/test_update.py

```diff
@@ -0,0 +1,172 @@
+
+    update_firmware = AsyncMock()
+    await setup_integration(hass, None, update_firmware)
+
+    state = hass.states.get("update.friendly_name_firmware_update")
+
+    assert state
+    assert state.state == STATE_UNKNOWN
+    assert state.attributes[ATTR_DEVICE_CLASS] == UpdateDeviceClass.FIRMWARE
+    assert state.attributes[ATTR_INSTALLED_VERSION] is None
+    assert state.attributes[ATTR_LATEST_VERSION] is None
+    assert state.attributes[ATTR_RELEASE_URL] is None
+    assert state.attributes[ATTR_RELEASE_SUMMARY] is None
+    update_firmware.assert_not_called()
+
+
+async def test_on_latest_firmware(hass: HomeAssistant):
+    """Test device on latest firmware."""
+
+    update_firmware = AsyncMock()
+    await setup_integration(hass, LATEST_FIRMWARE_INSTALLED, update_firmware)
+
+    state = hass.states.get("update.friendly_name_firmware_update")
+
+    assert state
+    assert state.state == STATE_UNKNOWN
+    assert state.attributes[ATTR_DEVICE_CLASS] == UpdateDeviceClass.FIRMWARE
+    assert state.attributes[ATTR_INSTALLED_VERSION] == "4.100.502"
+    assert state.attributes[ATTR_LATEST_VERSION] is None
+    assert state.attributes[ATTR_RELEASE_URL] is None
+    assert state.attributes[ATTR_RELEASE_SUMMARY] is None
+    update_firmware.assert_not_called()
+
+
+async def test_update_available(hass: HomeAssistant):
+    """Test device has firmware update available."""
+
+    update_firmware = AsyncMock()
+    await setup_integration(hass, FIRMWARE_UPDATE_AVAILABLE, update_firmware)
+
+    state = hass.states.get("update.friendly_name_firmware_update")
+
+    assert state
+    assert state.state == STATE_ON
+    assert state.attributes[ATTR_DEVICE_CLASS] == UpdateDeviceClass.FIRMWARE
+    assert state.attributes[ATTR_INSTALLED_VERSION] == "4.99.491"
+    assert state.attributes[ATTR_LATEST_VERSION] == "4.100.502"
+    assert (
+        state.attributes[ATTR_RELEASE_URL]
+        == "http://docs.linn.co.uk/wiki/index.php/ReleaseNotes"
+    )
+    assert (
+        state.attributes[ATTR_RELEASE_SUMMARY]
+        == "Release build version 4.100.502 (07 Jun 2023 12:29:48)"
+    )
+
+    await hass.services.async_call(
+        PLATFORM_DOMAIN,
+        SERVICE_INSTALL,
+        {ATTR_ENTITY_ID: "update.friendly_name_firmware_update"},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+
+    update_firmware.assert_called_once()
+
+
+async def test_firmware_update_not_required(hass: HomeAssistant):
+    """Ensure firmware install does nothing if up to date."""
+
+    update_firmware = AsyncMock()
+    await setup_integration(hass, LATEST_FIRMWARE_INSTALLED, update_firmware)
+
+    with pytest.raises(HomeAssistantError):
+        await hass.services.async_call(
+            PLATFORM_DOMAIN,
+            SERVICE_INSTALL,
+            {ATTR_ENTITY_ID: "update.friendly_name_firmware_update"},
+            blocking=True,
+        )
+        update_firmware.assert_not_called()
```

Comment:
Since the call before it raises, you never tested this assertion.
```suggestion
    update_firmware.assert_not_called()
```